### ISSUE TITLE
Vendor and Product ID are ignored if specified in decimal format in the xml file.

### ISSUE THREAD

**Issue by tomazsustar** [synapticon/siitool#12]
In that case only zeros are written to the bin file.

**Comment by fjes** [comment]
@tomazsustar can you provide a example snippet of the entries which fail?

For example our Vendor ID is written as

```
<EtherCATInfo Version="1.2">
  <Vendor>
    <Id>#x000022D2</Id>
    <Name>Synapticon GmbH</Name>
...
```

Also the product code is defined as hex value

```
...
   <Devices>
      <Device Physics="YY">
        <Type ProductCode="#x00000201" RevisionNo="#x0a000002">CiA402 Motor Control Device</Type>
        <Name LcId="1033">SOMANET CiA402 Drive</Name>
...
```
Both values are read and used.

**Comment by tomazsustar** [comment]
For instance this file from Beckoff has vendor id in decimal format. When reading it with siitool it will say that the vendor ID is 0. It will also put 0 for the vendor ID in the generated bin file.
```
<?xml version="1.0"?>
<EtherCATInfo xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="EtherCATInfo.xsd" Version="1.2">
	<Vendor>
		<Id>2</Id>
		<Name>Beckhoff Automation GmbH &amp; Co. KG</Name>
		...
	</Vendor>
	<Descriptions>
		<Groups>
			<Group SortOrder="310">
				<Type>DigIn</Type>
				<Name LcId="1033">Digital Input Terminals (EL1xxx)</Name>
				<Name LcId="1031">Digitale Eingangsklemmen (EL1xxx)</Name>
				<ImageData16x14>
```

The ESI file specification states that both formats are valid for HexDecValue type. 

So basically wherever one can specify a value as "#x12345" in hexadecimal format he should also be able to specify a value in decimal format e.g. "12345". I proposed a change in pull request https://github.com/synapticon/siitool/pull/13 which allows both formats.

**Comment by fjes** [comment]
I see, yes that's a bug, thank you for reporting.

**Comment by fjes** [comment]
Forgot to mention here, merged: 6f8904c013d37120618cf9ba9c0f5118ae407220
